protocol/x/clob/keeper/deleveraging.go

```diff
@@ -305,22 +329,50 @@ func (k Keeper) OffsetSubaccountPerpetualPosition(
 	return fills, deltaQuantumsRemaining
 }
 
+// getDeleveragingQuoteQuantums returns the quote quantums delta to apply to a deleveraging operation.
+// This returns the bankruptcy price for standard deleveraging operations, and the oracle price for
+// final settlement deleveraging operations. The type of deleveraging event is determined by the
+// clob pair status of the clob pair associated with the provided perpetual.
+func (k Keeper) getDeleveragingQuoteQuantumsDelta(
+	ctx sdk.Context,
+	perpetualId uint32,
+	subaccountId satypes.SubaccountId,
+	deltaQuantums *big.Int,
+) (deltaQuoteQuantums *big.Int, err error) {
+	clobPair := k.mustGetClobPairForPerpetualId(ctx, perpetualId)
```

Comment:
Question: is it okay if we panic here? This logic is executed in PrepareCheckState. Would we rather recover from this and continue? I think it is fine to panic but just want to double check.

Reply:
yeah should be fine to panic - if clob pair is missing, then something is really really wrong

Reply:
Just realized this function isn't doing exactly what I want for usability. The purpose of this function is to allow the deleveraging flow to be reusable such that non-negative TNC subaccounts with positions in a final settlement market can be closed at the oracle price.

Currently this function is written such that ALL subaccounts (including negative TNC) for the closed market will be settled at oracle price. This isn't what we want. I think the easiest way to updated this would be to do a collateral check and use the oracle price iff `isFinalSettlement && nonNegativeTNC`. 

I think what I'll do here is use `CanDeleverageSubaccount` to determine if the account has non negative TNC and only call this if the market is in final settlement.